src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1598,6 +1601,18 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         RsDiagnostic.AwaitOutsideAsyncContext(element, fix).addToHolder(holder)
     }
 
+    private fun checkTypeParameterList(holder: RsAnnotationHolder, path: RsPath) {
+        val resolved = path.reference?.resolve() ?: return
+        val typeParameter = path.contextOfType<RsPathType>()?.owner?.context as? RsTypeParameter ?: return
```

Comment:
I'd also swapped these lines because the second line will more likely lead to early return 
```suggestion
        val typeParameter = path.contextOfType<RsPathType>()?.owner?.context as? RsTypeParameter ?: return
        val resolved = path.reference?.resolve() ?: return
```